default/AI/ResourcesAI.py

```diff
@@ -29,98 +38,258 @@
-        statusStr += " Focus: % 8s" % ("_".join(str(planet.focus).split("_")[1:])[:8]) + " Species: " + str(planet.speciesName) + " Pop: %2d/%2d" % (planetPopulation, maxPop)
-        print statusStr
-    print "\n\nEmpire Totals:\nPopulation: %5d \nProduction: %5d\nResearch: %5d\n" % (empire.population(), empire.productionPoints, empire.resourceProduction(fo.resourceType.research))
-    if warnings != {}:
-        for pname in warnings:
-            mp, cp = warnings[pname]
-            print "Population Warning! -- %s has unsustainable current pop %d -- target %d" % (pname, cp, mp)
+        self.possible_output[self.current_focus] = (itarget, rtarget)
+        self.future_focus = self.current_focus
+
+
+class PlanetFocusManager(object):
+    """PlanetFocusManager tracks all of the empire's planets, what their current and future focus will be."""
+
+    def __init__(self):
+        universe = fo.getUniverse()
+
+        resource_timer.start("getPlanets")
+        planet_ids = list(PlanetUtilsAI.get_owned_planets_by_empire(universe.planetIDs))
+
+        # resource_timer.start("Shuffle")
+        # shuffle(generalPlanetIDs)
+
+        resource_timer.start("Targets")
+        planet_infos = [PlanetFocusInfo(universe.getPlanet(pid)) for pid in  planet_ids]
+        self.all_planet_info = {}
+        self.all_planet_info.update(zip(planet_ids, planet_infos))
+
+        self.raw_planet_info = dict(self.all_planet_info)
+        self.baked_planet_info = dict()
+
+        for pid, pinfo in self.raw_planet_info.items():
+            focusable = pinfo.planet.currentMeterValue(fo.meterType.targetPopulation) > 0
+            if not focusable:
+                self.baked_planet_info[pid] = self.raw_planet_info.pop(pid)
+
+
+    def bake_future_focus(self, pid, focus, update=True):
+        """Set the focus and moves it from the raw list to the baked list of planets
+        Return success or failure
+        """
+        pinfo = self.raw_planet_info.get(pid)
+        success = (pinfo is not None and
+                   (pinfo.current_focus == focus
+                    or (focus in pinfo.planet.availableFoci
+                        and fo.issueChangeFocusOrder(pid, focus))))
+        if success:
+            pinfo.future_focus = focus
+            self.baked_planet_info[pid] = self.raw_planet_info.pop(pid)
+
+            if update and pinfo.current_focus != focus:
+                # TODO determine if this should update all planets (for reporting) or just the remaining unplaced planets
+                universe = fo.getUniverse()
+                universe.updateMeterEstimates(self.raw_planet_info.keys())
+                itarget = pinfo.planet.currentMeterValue(fo.meterType.targetIndustry)
+                rtarget = pinfo.planet.currentMeterValue(fo.meterType.targetResearch)
+                pinfo.possible_output[focus] = (itarget, rtarget)
```

Comment:
You are correct.
It was intentional, but then I moved the code to grab the meterValues for the current planet to after the point where pinfo was transferred.